src/System.Numerics.Tensors/System/Numerics/Tensor.cs

```diff
@@ -1458,19 +1458,19 @@ public virtual Tensor<T> Slice(ReadOnlySpan<Range> ranges)
                 Debug.Assert(range.Length > 0);
 
                 int currentDimension = Dimensions[i];
-                if ((uint)range.Index >= (uint)currentDimension)
+                if ((uint)range.First >= (uint)currentDimension)
                 {
-                    throw new ArgumentOutOfRangeException($"Range.Index at position '{i}' must be non-negative and less than the Dimension '{currentDimension}' at that position. The Range.Index value is '{range.Index}'.", nameof(ranges));
+                    throw new ArgumentOutOfRangeException($"Range.Index at position '{i}' must be non-negative and less than the Dimension '{currentDimension}' at that position. The Range.Index value is '{range.First}'.", nameof(ranges));
```

Comment:
There are also usages of `Range.Index` in the string that should be replaced here and below.

Reply:
will fix